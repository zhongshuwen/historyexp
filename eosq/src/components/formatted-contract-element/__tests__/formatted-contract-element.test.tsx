import { shallow } from "enzyme"
import * as React from "react"
import { FormattedContractElement } from "../formatted-contract-element"

describe("formatted-contract-element", () => {
  it("correctly renders date when type is time_point", () => {
    expect(
      shallow(
        <FormattedContractElement type="time_point" value="2018-07-30T12:52:54.000" label="" />
      )
    ).toMatchSnapshot()
  })

  it("correctly renders date when type is time_point_sec", () => {
    expect(
      shallow(
        <FormattedContractElement type="time_point_sec" value="2018-07-30T12:52:54" label="" />
      )
    ).toMatchSnapshot()
  })

  it("correctly renders date when type is string as number in seconds", () => {
    expect(
      shallow(<FormattedContractElement type="string" value="1536262055" label="created_at" />)
    ).toMatchSnapshot()
  })

  it("correctly renders date when type is string as number in milliseconds", () => {
    expect(
      shallow(<FormattedContractElement type="string" value="1536262086756" label="created_at" />)
    ).toMatchSnapshot()
  })
})
