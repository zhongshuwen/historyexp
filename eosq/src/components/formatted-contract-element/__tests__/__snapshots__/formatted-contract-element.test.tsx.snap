// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatted-contract-element correctly renders date when type is string as number in milliseconds 1`] = `
<Styled(div)
  color="text"
>
  06 Sep 2018 07:28
</Styled(div)>
`;

exports[`formatted-contract-element correctly renders date when type is string as number in seconds 1`] = `
<Styled(div)
  color="text"
>
  06 Sep 2018 07:27
</Styled(div)>
`;

exports[`formatted-contract-element correctly renders date when type is time_point 1`] = `
<Styled(div)
  color="text"
>
  2018-07-30T12:52:54 UTC
</Styled(div)>
`;

exports[`formatted-contract-element correctly renders date when type is time_point_sec 1`] = `
<Styled(div)
  color="text"
>
  2018-07-30T12:52:54 UTC
</Styled(div)>
`;
